From ecf39d32f968e31c94a708fc8143eb7208e75e72 Mon Sep 17 00:00:00 2001
From: Brian McKenna <brian@brianmckenna.org>
Date: Sun, 7 Aug 2022 08:39:26 +1000
Subject: [PATCH] Fallback to screensaver inhibit via D-Bus on Wayland

---
 gfx/common/wayland_common.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/gfx/common/wayland_common.c b/gfx/common/wayland_common.c
index 1a3a443b28..283da225f9 100644
--- a/gfx/common/wayland_common.c
+++ b/gfx/common/wayland_common.c
@@ -226,6 +226,12 @@ void gfx_ctx_wl_destroy_resources_common(gfx_ctx_wayland_data_t *wl)
       xdg_toplevel_destroy(wl->xdg_toplevel);
    if (wl->idle_inhibit_manager)
       zwp_idle_inhibit_manager_v1_destroy(wl->idle_inhibit_manager);
+   else {
+#ifdef HAVE_DBUS
+      dbus_screensaver_uninhibit();
+      dbus_close_connection();
+#endif
+   }
    if (wl->deco)
       zxdg_toplevel_decoration_v1_destroy(wl->deco);
    if (wl->deco_manager)
@@ -520,6 +526,9 @@ bool gfx_ctx_wl_init_common(
    if (!wl->idle_inhibit_manager)
    {
       RARCH_LOG("[Wayland]: Compositor doesn't support zwp_idle_inhibit_manager_v1 protocol\n");
+#ifdef HAVE_DBUS
+      dbus_ensure_connection();
+#endif
    }
 
    if (!wl->deco_manager)
@@ -707,8 +716,15 @@ bool gfx_ctx_wl_suppress_screensaver(void *data, bool state)
 {
    gfx_ctx_wayland_data_t *wl = (gfx_ctx_wayland_data_t*)data;
 
-   if (!wl->idle_inhibit_manager)
+   if (!wl->idle_inhibit_manager) {
+#ifdef HAVE_DBUS
+      /* Some Wayland compositors (e.g. Phoc) don't implement Wayland's Idle protocol.
+       * They instead rely on things like Gnome Screensaver. */
+      return dbus_suspend_screensaver(state);
+#else
       return false;
+#endif
+   }
    if (state == (!!wl->idle_inhibitor))
       return true;
 
-- 
2.36.1

